shared.js tests

zZeck opened the pull request:
Back again. Hope you are doing well.
Just lemme know if I need to change anything up.

A few questions:
1. Do these tests look ok for the functions they cover? Too much? Not enough?
2. I made a couple of comments in the regular code to indicate valid states for data. Are they ok with you?
3. Is the getTransform function name a typo? To me, it is setting something. Maybe I misunderstand.
4. setRenderTarget seems to have unused locals. Should I remove them?

Commit:
shared.js tests

JohnEarnest commented:
1) There isn't anything inherently wrong with these tests, but I will say that it seems like a lot of testing code for a small and fairly stable corner of the codebase. For any test, the benefit of coverage must be weighed against the cost of updating the test in the future to reflect changes to the codebase. Sometimes careful code review and cleanup can be just as beneficial, as in your item (4).
2) This is fine. In the case of `screenRotation` it might be better to introduce an assertion in `unpackOptions()`. Per the current behavior, values outside {90,180,270} should be coerced to 0.
3) I don't have a strong opinion about the name of this function.
4) `hm`/`wm` look like cruft. Go ahead and remove them.

zZeck commented:
Yeah, I definitely delete tests if/when they get in the road.  No attachment. Sometimes larger modules have their own tests and are being internally restructured.  Sometimes, the tests just don't matter for other reasons. Tests are a good way to go over codebase in focused way for me.

Added an assertion to that function. Screen rotation looks like it is referenced in a couple other places, so I left the comment thinking somebody would probably look there first for info.  

Renamed function.

Removed unused locals.

Manually tested octo and all the rotation stuff works still.
